SwiftParserCLI/Sources/swift-parser-cli/ParseCommand.swift

```diff
@@ -50,4 +59,52 @@ extension ParseCommand {
 
   /// Whether sequence folding using standard operators should be performed
   var foldSequences: Bool { arguments.foldSequences }
+
+  /// Parse the source file, applying any additional configuration options
+  /// such as sequence folding, and provide it to the given closure.
+  func withParsedSourceFile<R>(
+    wantDiagnostics: Bool = true,
+    body: (SourceFileSyntax, [Diagnostic]) throws -> R
+  ) throws -> R {
+    return try sourceFileContents.withUnsafeBufferPointer { sourceBuffer in
```

Comment:
Since we intern the source buffer into the syntax arena, I think you could just return `(SourceFileSyntax, [Diagnostic])` here and avoid the trip through the `body` closure. That should make the code on the call side look a little more straight-line, I think.